src/InsetOrderOptimizer.cpp

```diff
@@ -197,20 +197,17 @@ BinJunctions InsetOrderOptimizer::variableWidthPathToBinJunctions(const Variable
             // Sort into the right bin, ...
             size_t bin_index;
             const bool in_hole_region = line.region_id != outer_region_id && line.region_id != 0;
-            if(ignore_inner_inset_order)
+            if(center_last && line.is_odd)
+            {
+                bin_index = 0;
+            }
+            else if(pack_regions_by_inset)
             {
-                bin_index = std::min(inset_index, static_cast<size_t>(1)) + 2 * (in_hole_region ? line.region_id : 0);
+                bin_index = std::min(inset_index, static_cast<size_t>(1)) + 2 * (in_hole_region ? line.region_id : 0) + center_last;
```

Comment:
Side Note: Though I'm not sure where we stand (as a team or company) re:code-style on implicit bool conversion like you did with `... + center_last`, I personally think it's fine. After all, it's both readable, and _explicitly_ in the C++ standard that `false = 0` and `true = 1` :-)

Reply:
I'm fine with changing this as well. I agree it's a bit hard to read.